### ISSUE TITLE
User-defined attributes?

### ISSUE THREAD

**Issue by DethRaid** [xoofx/CppAst.NET#39]
Does this library supports parsing user-defined C++11 attributes? I was hoping to annotate some of my C++ code with them and generate C# bindings from classes with specific attributes. I just ran a quick test on my code and this library reports no attributes.

https://github.com/DethRaid/SanityEngine/blob/trunk/SanityEngine/src/sanity_engine.hpp#L21 is the class I'm trying to generate bindings for, https://github.com/DethRaid/SanityEngine/blob/trunk/SanityEngine.Codegen.NET/CodegenProgram.cs#L45 is where I'm checking for my attribute.

I've inspected the `CppAst.ClassCompilation` for my class in the Visual Studio debugger - there's no attributes, or really much of anything. Is there something that's making my class really hard to parse?

**Comment by xoofx** [comment]
I don't think that we are parsing attributes for C++ class so they are not reported. They are parsed for methods and arguments afair. I have no personal time to work on this but a PR is welcome.
